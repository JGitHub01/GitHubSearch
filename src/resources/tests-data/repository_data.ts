export const search_result = {
  "keyword": "tetri",
  "isSearching": true,
  "total_count": 40,
  "incomplete_results": false,
  "items": [
    {
      "id": 343937425,
      "node_id": "MDEwOlJlcG9zaXRvcnkzNDM5Mzc0MjU=",
      "name": "Tetrio-Rank-Predictor",
      "full_name": "chunkanglu/Tetrio-Rank-Predictor",
      "private": false,
      "owner": {
        "login": "chunkanglu",
        "id": 75538064,
        "node_id": "MDQ6VXNlcjc1NTM4MDY0",
        "avatar_url": "https://avatars.githubusercontent.com/u/75538064?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/chunkanglu",
        "html_url": "https://github.com/chunkanglu",
        "followers_url": "https://api.github.com/users/chunkanglu/followers",
        "following_url": "https://api.github.com/users/chunkanglu/following{/other_user}",
        "gists_url": "https://api.github.com/users/chunkanglu/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/chunkanglu/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/chunkanglu/subscriptions",
        "organizations_url": "https://api.github.com/users/chunkanglu/orgs",
        "repos_url": "https://api.github.com/users/chunkanglu/repos",
        "events_url": "https://api.github.com/users/chunkanglu/events{/privacy}",
        "received_events_url": "https://api.github.com/users/chunkanglu/received_events",
        "type": "User",
        "site_admin": false
      },
      "html_url": "https://github.com/chunkanglu/Tetrio-Rank-Predictor",
      "description": "Provides an approximate rank for Tetr.io's Tetra League based on APM, PPS, and VS-Score.",
      "fork": false,
      "url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor",
      "forks_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/forks",
      "keys_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/keys{/key_id}",
      "collaborators_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/collaborators{/collaborator}",
      "teams_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/teams",
      "hooks_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/hooks",
      "issue_events_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/issues/events{/number}",
      "events_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/events",
      "assignees_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/assignees{/user}",
      "branches_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/branches{/branch}",
      "tags_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/tags",
      "blobs_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/git/blobs{/sha}",
      "git_tags_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/git/tags{/sha}",
      "git_refs_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/git/refs{/sha}",
      "trees_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/git/trees{/sha}",
      "statuses_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/statuses/{sha}",
      "languages_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/languages",
      "stargazers_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/stargazers",
      "contributors_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/contributors",
      "subscribers_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/subscribers",
      "subscription_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/subscription",
      "commits_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/commits{/sha}",
      "git_commits_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/git/commits{/sha}",
      "comments_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/comments{/number}",
      "issue_comment_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/issues/comments{/number}",
      "contents_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/contents/{+path}",
      "compare_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/compare/{base}...{head}",
      "merges_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/merges",
      "archive_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/{archive_format}{/ref}",
      "downloads_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/downloads",
      "issues_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/issues{/number}",
      "pulls_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/pulls{/number}",
      "milestones_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/milestones{/number}",
      "notifications_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/notifications{?since,all,participating}",
      "labels_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/labels{/name}",
      "releases_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/releases{/id}",
      "deployments_url": "https://api.github.com/repos/chunkanglu/Tetrio-Rank-Predictor/deployments",
      "created_at": "2021-03-02T23:04:08Z",
      "updated_at": "2021-03-08T20:29:00Z",
      "pushed_at": "2021-03-08T20:28:58Z",
      "git_url": "git://github.com/chunkanglu/Tetrio-Rank-Predictor.git",
      "ssh_url": "git@github.com:chunkanglu/Tetrio-Rank-Predictor.git",
      "clone_url": "https://github.com/chunkanglu/Tetrio-Rank-Predictor.git",
      "svn_url": "https://github.com/chunkanglu/Tetrio-Rank-Predictor",
      "homepage": null,
      "size": 9015,
      "stargazers_count": 0,
      "watchers_count": 0,
      "language": "Python",
      "has_issues": true,
      "has_projects": true,
      "has_downloads": true,
      "has_wiki": true,
      "has_pages": false,
      "forks_count": 0,
      "mirror_url": null,
      "archived": false,
      "disabled": false,
      "open_issues_count": 0,
      "license": null,
      "forks": 0,
      "open_issues": 0,
      "watchers": 0,
      "default_branch": "main",
      "score": 1
    },
    {
      "id": 324767424,
      "node_id": "MDEwOlJlcG9zaXRvcnkzMjQ3Njc0MjQ=",
      "name": "Cooperative-Protein-Dynamics",
      "full_name": "research555/Cooperative-Protein-Dynamics",
      "private": false,
      "owner": {
        "login": "research555",
        "id": 62707744,
        "node_id": "MDQ6VXNlcjYyNzA3NzQ0",
        "avatar_url": "https://avatars.githubusercontent.com/u/62707744?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/research555",
        "html_url": "https://github.com/research555",
        "followers_url": "https://api.github.com/users/research555/followers",
        "following_url": "https://api.github.com/users/research555/following{/other_user}",
        "gists_url": "https://api.github.com/users/research555/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/research555/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/research555/subscriptions",
        "organizations_url": "https://api.github.com/users/research555/orgs",
        "repos_url": "https://api.github.com/users/research555/repos",
        "events_url": "https://api.github.com/users/research555/events{/privacy}",
        "received_events_url": "https://api.github.com/users/research555/received_events",
        "type": "User",
        "site_admin": false
      },
      "html_url": "https://github.com/research555/Cooperative-Protein-Dynamics",
      "description": "This is a simple program that maps the dynamics between LacI and TetR in a one of the first toggle switch systems created by Gardner et al (2000)",
      "fork": false,
      "url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics",
      "forks_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/forks",
      "keys_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/keys{/key_id}",
      "collaborators_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/collaborators{/collaborator}",
      "teams_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/teams",
      "hooks_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/hooks",
      "issue_events_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/issues/events{/number}",
      "events_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/events",
      "assignees_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/assignees{/user}",
      "branches_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/branches{/branch}",
      "tags_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/tags",
      "blobs_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/git/blobs{/sha}",
      "git_tags_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/git/tags{/sha}",
      "git_refs_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/git/refs{/sha}",
      "trees_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/git/trees{/sha}",
      "statuses_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/statuses/{sha}",
      "languages_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/languages",
      "stargazers_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/stargazers",
      "contributors_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/contributors",
      "subscribers_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/subscribers",
      "subscription_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/subscription",
      "commits_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/commits{/sha}",
      "git_commits_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/git/commits{/sha}",
      "comments_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/comments{/number}",
      "issue_comment_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/issues/comments{/number}",
      "contents_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/contents/{+path}",
      "compare_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/compare/{base}...{head}",
      "merges_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/merges",
      "archive_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/{archive_format}{/ref}",
      "downloads_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/downloads",
      "issues_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/issues{/number}",
      "pulls_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/pulls{/number}",
      "milestones_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/milestones{/number}",
      "notifications_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/notifications{?since,all,participating}",
      "labels_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/labels{/name}",
      "releases_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/releases{/id}",
      "deployments_url": "https://api.github.com/repos/research555/Cooperative-Protein-Dynamics/deployments",
      "created_at": "2020-12-27T13:27:35Z",
      "updated_at": "2020-12-30T14:58:26Z",
      "pushed_at": "2020-12-30T14:58:24Z",
      "git_url": "git://github.com/research555/Cooperative-Protein-Dynamics.git",
      "ssh_url": "git@github.com:research555/Cooperative-Protein-Dynamics.git",
      "clone_url": "https://github.com/research555/Cooperative-Protein-Dynamics.git",
      "svn_url": "https://github.com/research555/Cooperative-Protein-Dynamics",
      "homepage": null,
      "size": 6,
      "stargazers_count": 0,
      "watchers_count": 0,
      "language": "Python",
      "has_issues": true,
      "has_projects": true,
      "has_downloads": true,
      "has_wiki": true,
      "has_pages": false,
      "forks_count": 0,
      "mirror_url": null,
      "archived": false,
      "disabled": false,
      "open_issues_count": 0,
      "license": null,
      "forks": 0,
      "open_issues": 0,
      "watchers": 0,
      "default_branch": "main",
      "score": 1
    }
  ]
};
