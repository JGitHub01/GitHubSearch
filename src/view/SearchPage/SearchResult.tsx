import { Link } from 'react-router-dom';
import { ISearchResult } from 'lib/models';
import RepositoryItem from './RepositoryItem';

interface ISearchResultProps extends ISearchResult {
  keyword?: string
  isSearching?: boolean
}
export default function SearchResult(props: ISearchResultProps) {
  const {
    keyword,
    isSearching = false,
    total_count,
    items,
    success = true,
    msg
  } = props;
  return (
    <div className='github-search-result' data-testid='search-result'>
      {(total_count > 0 && items != null) ?
        <div className='github-search-result__list' key='list'>
          {items.map((r, idx) =>
            <Link className='detail-link'
              to={{ pathname: '/detail', state: { ...r } }}
              key={idx}
            >
              <RepositoryItem name={r.name} description={r.description} />
            </Link>)
          }
        </div> :
        <div>
          {!success ? `Trouble search GitHub: ${msg}` :
            isSearching ? 'Searching GitHub repositoies ...' :
              keyword ? 'No repositories are found!' : 'Please enter a keyword!'}
        </div>
      }
    </div>
  );
}
