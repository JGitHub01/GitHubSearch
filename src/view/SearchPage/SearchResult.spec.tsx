import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import SearchResult from './SearchResult';
import { search_result } from 'resources/tests-data/repository_data';

describe('SearchResult', () => {
  it('Should render the basic snapshot view with no keyword', () => {
    const { getByTestId } = render(
      <Router>
        <SearchResult total_count={0} />
      </Router>
    );
    expect(getByTestId('search-result')).toMatchSnapshot();
  });
  it('Should render the snapshot view with no repository found', () => {
    const { getByTestId } = render(
      <Router>
        <SearchResult total_count={0} keyword="blabla"/>
      </Router>
    );
    expect(getByTestId('search-result')).toMatchSnapshot();
  });
  it('Should render the snapshot view of successful search', () => {
    const { getByTestId } = render(
      <Router>
        <SearchResult {...search_result} />
      </Router>
    );
    expect(getByTestId('search-result')).toMatchSnapshot();
  });
  it('Should render the snapshot view of failed search', () => {
    const failed_search = {
      total_count: 0,
      success: false,
      msg: "Network failure!"
    }
    const { getByTestId } = render(
      <Router>
        <SearchResult {...failed_search} />
      </Router>
    );
    expect(getByTestId('search-result')).toMatchSnapshot();
  });
});
