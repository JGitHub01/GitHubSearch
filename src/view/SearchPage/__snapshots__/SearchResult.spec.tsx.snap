// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult Should render the basic snapshot view with no keyword 1`] = `
<div
  class="github-search-result"
  data-testid="search-result"
>
  <div>
    Please enter a keyword!
  </div>
</div>
`;

exports[`SearchResult Should render the snapshot view of failed search 1`] = `
<div
  class="github-search-result"
  data-testid="search-result"
>
  <div>
    Trouble search GitHub: Network failure!
  </div>
</div>
`;

exports[`SearchResult Should render the snapshot view of successful search 1`] = `
<div
  class="github-search-result"
  data-testid="search-result"
>
  <div
    class="github-search-result__list"
  >
    <a
      class="detail-link"
      href="/detail"
    >
      <div
        class="repository"
        data-testid="repository-item"
      >
        <div
          class="repository-name"
        >
          <span
            class="repository-name__label"
          >
            Name:
          </span>
          <div
            class="repository-name__text"
          >
            Tetrio-Rank-Predictor
          </div>
        </div>
        <div
          class="repository-description"
        >
          <span
            class="repository-description__label"
          >
            Description:
          </span>
          <div
            class="repository-description__text"
          >
            Provides an approximate rank for Tetr.io's Tetra League based on APM, PPS, and VS-Score.
          </div>
        </div>
      </div>
    </a>
    <a
      class="detail-link"
      href="/detail"
    >
      <div
        class="repository"
        data-testid="repository-item"
      >
        <div
          class="repository-name"
        >
          <span
            class="repository-name__label"
          >
            Name:
          </span>
          <div
            class="repository-name__text"
          >
            Cooperative-Protein-Dynamics
          </div>
        </div>
        <div
          class="repository-description"
        >
          <span
            class="repository-description__label"
          >
            Description:
          </span>
          <div
            class="repository-description__text"
          >
            This is a simple program that maps the dynamics between LacI and TetR in a one of the first toggle switch systems created by Gardner et al (2000)
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`SearchResult Should render the snapshot view with no repository found 1`] = `
<div
  class="github-search-result"
  data-testid="search-result"
>
  <div>
    No repositories are found!
  </div>
</div>
`;
